tensorflow/compiler/xla/service/gpu/tests/tree_reduction_rewriter_test.cc

```diff
@@ -70,7 +70,7 @@ ENTRY main {
 )";
 
   // TODO(cheshire): a more generic check, do not hardcode the names.
-  MatchOptimizedHloWithShapes(hlo_text,
+  /*  MatchOptimizedHloWithShapes(hlo_text,
```

Comment:
You probably missed that this test and another one below are still commented out.

Reply:
This PR isn't ready to merge, just ready to benchmarks. As it isn't sure we will merge it as is, I didn't fix the tests. We need to figure out if this PR hurts the tree reductions. If so, I can make a slightly more complication version that won't hurt it. But @cheshire isn't sure the complication is useful. So he wanted to benchmark it first.

Reply:
Done. Maybe the interface can be simplifier, but before doing it, I would like to know if I finish it like this or I make it a little bit more complicated to not slowdown the tree reduction.